apps/examples/testcase/ta_tc/arastorage/itc/itc_arastorage_main.c

```diff
@@ -56,127 +55,665 @@ struct arastorage_data_type_s {
+* @scenario             check initialization and deinitialization database operation returns DB_OK
+* @apicovered           db_deinit, db_init
+* @precondition         none
+* @postcondition        none
+*/
+void itc_arastorage_db_init_deinit_p(void)
+{
+	db_result_t ret;
+
+	ret = db_init();
+	TC_ASSERT_EQ("db_init", ret, DB_OK);
+
+	ret = db_deinit();
+	TC_ASSERT_EQ("db_deinit", ret, DB_OK);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase             itc_arastorage_db_init_deinit_p_multi_time
+* @brief                To check initialization and deinitialization database resources
+* @scenario             check iterative initialization and deinitialization database operation returns DB_OK
+* @apicovered           db_deinit, db_init
+* @precondition         none
+* @postcondition        none
+*/
+void itc_arastorage_db_init_deinit_p_multi_time(void)
+{
+	db_result_t ret;
+	int iteration = 0;
+
+	for (iteration = 0; iteration < 10; ++iteration) {
+		ret = db_init();
+		TC_ASSERT_EQ("db_init", ret, DB_OK);
+
+		ret = db_deinit();
+		TC_ASSERT_EQ("db_deinit", ret, DB_OK);
+	}
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase             itc_arastorage_db_print_header_n_after_deinit
+* @brief                To check print of the related information : relation, attribute name
+* @scenario             If used api returns DB_OK
+* @apicovered           db_print_header, db_query
+* @precondition         Database resource must be deinitialized
+* @postcondition        none
+*/
+void itc_arastorage_db_print_header_n_after_deinit(void)
+{
+	db_result_t ret;
+	const char *rel_name = RELATION_NAME2;
+
+	ret = db_init();
+	TC_ASSERT_EQ("db_init", ret, DB_OK);
+
+	memset(g_query, 0, QUERY_LENGTH);
+	snprintf(g_query, QUERY_LENGTH, "CREATE RELATION %s;", rel_name);
+	ret = db_exec(g_query);
+	TC_ASSERT_EQ("db_exec", ret, DB_OK);
+
+	memset(g_query, 0, QUERY_LENGTH);
+	snprintf(g_query, QUERY_LENGTH, "CREATE ATTRIBUTE %s DOMAIN int IN %s;", g_attribute_set[0], rel_name);
+	ret = db_exec(g_query);
+	TC_ASSERT_EQ_CLEANUP("db_exec", ret, DB_OK, db_cleanup(rel_name); db_deinit());
+
+	memset(g_query, 0, QUERY_LENGTH);
+	snprintf(g_query, QUERY_LENGTH, "CREATE ATTRIBUTE %s DOMAIN long IN %s;", g_attribute_set[1], rel_name);
+	ret = db_exec(g_query);
+	TC_ASSERT_EQ_CLEANUP("db_exec", ret, DB_OK, db_cleanup(rel_name); db_deinit());
+
+	memset(g_query, 0, QUERY_LENGTH);
+	snprintf(g_query, QUERY_LENGTH, "CREATE ATTRIBUTE %s DOMAIN string(32) IN %s;", g_attribute_set[2], rel_name);
+	ret = db_exec(g_query);
+	TC_ASSERT_EQ_CLEANUP("db_exec", ret, DB_OK, db_cleanup(rel_name); db_deinit());
+
+	memset(g_query, 0, QUERY_LENGTH);
+	sprintf(g_query, "CREATE INDEX %s.%s TYPE %s;", rel_name, g_attribute_set[0], RELATION_INDEX);
+	ret = db_exec(g_query);
+	TC_ASSERT_EQ_CLEANUP("db_exec", ret, DB_OK, db_cleanup(rel_name); db_deinit());
+
+	int index;
+	for (index = 0; index < DATA_SET_NUM; index++) {
```

Comment:
There are many duplcated loop code in this file, why don't you make functions for these?
insert_blah_blah() ...